src/turing_model.jl

```diff
@@ -22,7 +22,17 @@ This will be expanded to `x1 + x2 + x1:x2`, which, following the principle of hi
 the main effects must also be added along with the interaction effects. Here `x1:x2`
 means that the values of `x1` will be multiplied (interacted) with the values of `x2`.
 
-TODO: add random-effects.
+Random-effects (a.k.a. group-level effects) can be specified with the `(term | group)` inside
+the `@formula`, where `term` is the independent variable and `group` is the **categorical**
+representation (i.e. either a column of `String`s or a `CategoricalArray` in `data`) of the
+grouping variable.
```

Comment:
```suggestion
representation (i.e., either a column of `String`s or a `CategoricalArray` in `data`).
```
The last four words are correct, but make the sentence more difficult to read